mbtools/utils.py

```diff
@@ -56,12 +65,29 @@ def parse_question_bank_for_html(mbz_dir, ids=None):
     return html
 
 
-def find_references_containing(content_etree, src_content):
+def find_references_containing(content_etree, src_content=None):
     """
     Given a etree object and a prefix for a resource, this function will
     return the src values from the etree that contain the src_content prefix
     """
-    matching_elems = content_etree.xpath(
-        f'//*[contains(@src, "{src_content}")]'
-    )
+    matching_elems = []
+    if src_content is None:
+        matching_elems = content_etree.xpath(
+            '//*[@src]'
+        )
+    else:
+        matching_elems = content_etree.xpath(
+            f'//*[contains(@src, "{src_content}")]'
+        )
     return [el.get("src") for el in matching_elems]
+
+
+def replace_attribute_values_tree(content_tree, attrs, swap_mapping):
+    swaps = {}
+    for attr in attrs:
+        for elem in content_tree.xpath(f'//*[@{attr}]'):
+            im_filename = elem.attrib[attr]
+            if im_filename in swap_mapping.keys():
+                elem.attrib[attr] = swap_mapping[im_filename]
+                swaps[im_filename] = swap_mapping[im_filename]
+    return swaps
```

Comment:
The returned `swaps` is odd to me. It seems like callers use it to get a count of the number of changes, but a dict isn't going to convey that accurately due to collisions. I think this either needs to return an int or an array so the count is accurate.